flake/dev/packages.nix

```
@@ -3,16 +3,33 @@
   perSystem =
     { pkgs, config, ... }:
     {
-      # Build all packages with 'nix flake check' instead of only verifying they
-      # are derivations.
-      checks = config.packages;
-
       # Testbeds are virtual machines based on NixOS, therefore they are
       # only available for Linux systems.
-      packages = lib.mkIf pkgs.stdenv.hostPlatform.isLinux (
+      testbeds = lib.mkIf pkgs.stdenv.hostPlatform.isLinux (
         import ../../stylix/testbed {
           inherit pkgs inputs lib;
         }
       );
+
+      packages = config.testbeds;
+
+      # Build all packages with 'nix flake check' instead of only verifying they
+      # are derivations.
+      checks = config.packages;
+
+      ci.buildbot = lib.pipe config.testbeds [
+        (lib.mapAttrs' (
+          name: testbed:
+          let
+            # name is formatted as `testbed:target:variant` e.g. `testbed:alacritty:dark`
+            splitName = builtins.split ":" name;
+            target = builtins.elemAt splitName 2;
+            variant = builtins.elemAt splitName 4;
+          in
+          lib.nameValuePair target { ${variant} = testbed; }
+        ))
+        (lib.mapAttrs (target: testbeds: pkgs.linkFarm "testbeds-${target}" testbeds))
```

This can be simplified:

```suggestion
        (lib.mapAttrs (target: pkgs.linkFarm "testbeds-${target}"))
```